code/harry_potter_store/src/store.py

```diff
@@ -52,34 +52,33 @@ def get_price(self, to_buy_books: Dict[str, int]) -> float:
         """
         self._validate(to_buy_books)
         to_buy_books = {k: v for k, v in sorted(to_buy_books.items(), key=lambda item: item[1], reverse=True)}
-        discounts = [self._discount_wrapper(5, 0.25), self._discount_wrapper(4, 0.2),
-                     self._discount_wrapper(3, 0.1), self._discount_wrapper(2, 0.05)]
+        discounts = {2: self._discount_wrapper(2, 0.05), 3: self._discount_wrapper(3, 0.1),
+                     4: self._discount_wrapper(4, 0.2), 5: self._discount_wrapper(5, 0.25)}
         min_price = self.book_price * sum(to_buy_books.values())
         for i in range(len(discounts)):
```

Comment:
Why not make it simpler? You still don't need to have the second calculation inside iteration...
```python
    for key, discount in discounts.items():
            min_price = min(min_price,
                            self._get_price(to_buy_books, {key: discount}))
    min_price = min(min_price,
                        self._get_price(to_buy_books,
                                        dict(list(discounts.items())[::-1])))  # TODO: fix to not rely on dict order
    return min_price

```